native/packages/react-native-bpk-component-navigation-bar/src/BpkNavigationBar.android.js

```diff
@@ -0,0 +1,240 @@
+
+class BpkNavigationBar extends Component<Props, {}> {
+  theme: ?AndroidTheme;
+
+  static propTypes = {
+    title: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
+      .isRequired,
+    theme: makeThemePropType(ANDROID_THEME_ATTRIBUTES),
+    leadingButton: PropTypes.oneOf(Object.keys(BUTTON_TYPES)),
+    onLeadingButtonPress: PropTypes.func,
+    trailingButton: PropTypes.oneOf(Object.keys(BUTTON_TYPES)),
+    onTrailingButtonPress: PropTypes.func,
+    subtitleView: PropTypes.element,
+  };
+
+  static defaultProps = {
+    theme: null,
+    leadingButton: null,
+    onLeadingButtonPress: null,
+    trailingButton: null,
+    onTrailingButtonPress: null,
+    subtitleView: null,
+  };
+
+  constructor(props) {
+    super(props);
+    this.theme = getThemeStyle(
+      ANDROID_THEME_ATTRIBUTES,
+      this.props.theme || {},
+    );
+  }
+
+  componentDidMount() {
+    this.updateStatusBar();
+  }
+
+  componentDidUpdate() {
+    this.theme = getThemeStyle(
+      ANDROID_THEME_ATTRIBUTES,
+      this.props.theme || {},
+    );
+
+    this.updateStatusBar();
+  }
+
+  onLeadingButtonPress = () => {
+    if (this.props.onLeadingButtonPress) {
+      this.props.onLeadingButtonPress();
+    }
+  };
+
+  onTrailingButtonPress = () => {
+    if (this.props.onTrailingButtonPress) {
+      this.props.onTrailingButtonPress();
+    }
+  };
+
+  updateStatusBar() {
+    if (this.theme) {
+      const {
+        navigationBarStatusBarColor,
+        navigationBarStatusBarStyle,
+      } = this.theme;
+
+      StatusBar.setBackgroundColor(navigationBarStatusBarColor);
+      StatusBar.setBarStyle(navigationBarStatusBarStyle);
+    } else {
+      StatusBar.setBackgroundColor(colorBlue700);
+      StatusBar.setBarStyle('light-content');
+    }
+  }
+
+  render() {
+    const {
+      theme,
+      title,
+      leadingButton,
+      trailingButton,
+      subtitleView,
+    } = this.props;
+    const isTitleView = typeof title !== 'string';
+    const hasSubtitleView = subtitleView !== null;
+
+    const barStyle = [styles.bar];
+    const outerBarStyle = [styles.barOuter];
+    const titleStyle = [isTitleView ? styles.titleView : styles.titleString];
+    let tintColor = colorWhite;
+    let touchableColor = colorWhite;
+
+    if (theme) {
+      const { navigationBarTintColor, navigationBarBackgroundColor } = theme;
+
+      titleStyle.push({ color: navigationBarTintColor });
+      outerBarStyle.push({
+        backgroundColor: navigationBarBackgroundColor,
+      });
+      tintColor = navigationBarTintColor;
+      touchableColor = navigationBarTintColor;
+    }
+
+    if (hasSubtitleView) {
+      outerBarStyle.push(styles.barOuterWithSubtitle);
+    }
+
+    return (
+      <View style={outerBarStyle}>
+        <View style={barStyle}>
+          {leadingButton && (
+            <BpkNavigationBarButton
+              touchableColor={touchableColor}
+              type={leadingButton}
+              onPress={this.onLeadingButtonPress}
```

Comment:
The implementation for this just does:
```
onLeadingButtonPress = () => {
  if (this.props.onLeadingButtonPress) {
    this.props.onLeadingButtonPress();
  }
};
```
It can simply be `onPress={this.props.onLeadingButtonPress}`

Reply:
This has changed with the API overhaul